File: PHP/Komik.php
<?php
require_once "Buku.php";

class Komik extends Buku {
    private $genre;
    private $jumlahHalaman;
    private $volume;
    private $foto_produk;

    public function __construct($kodeProduk, $namaProduk, $harga, $penulis, $penerbit, $tahunTerbit,
                                $genre, $jumlahHalaman, $volume, $foto_produk) {
        parent::__construct($kodeProduk, $namaProduk, $harga, $penulis, $penerbit, $tahunTerbit);
        $this->genre = $genre;
        $this->jumlahHalaman = $jumlahHalaman;
        $this->volume = $volume;
        $this->foto_produk = $foto_produk;
    }

    public function getGenre() { return $this->genre; }
    public function getJumlahHalaman() { return $this->jumlahHalaman; }
    public function getVolume() { return $this->volume; }
    public function getFotoProduk() { return $this->foto_produk; }
}
?>

File: CPP/Komik.cpp
#include "Buku.cpp"
using namespace std;

class Komik : public Buku {
private:
    string genre;
    int jumlahHalaman;
    int volume;

public:
    Komik(string kodeProduk, string namaProduk, int harga,
          string penulis, string penerbit, int tahunTerbit,
          string genre, int jumlahHalaman, int volume)
        : Buku(kodeProduk, namaProduk, harga, penulis, penerbit, tahunTerbit) {
        this->genre = genre;
        this->jumlahHalaman = jumlahHalaman;
        this->volume = volume;
    }

    string getGenre() { return genre; }
    int getJumlahHalaman() { return jumlahHalaman; }
    int getVolume() { return volume; }

    void setGenre(string genre) { this->genre = genre; }
    void setJumlahHalaman(int jumlahHalaman) { this->jumlahHalaman = jumlahHalaman; }
    void setVolume(int volume) { this->volume = volume; }
};
